pkg/storage/snapshot/restore.go

```diff
@@ -1488,16 +1524,19 @@ func (ctrl *VMRestoreController) createRestorePVC(
 	if err != nil {
 		return err
 	}
+	if pvc.Annotations == nil {
+		pvc.Annotations = make(map[string]string)
+	}
 
 	if dvOwner != "" { // PVC is owned by a DV
-		if pvc.Annotations == nil {
-			pvc.Annotations = make(map[string]string)
-		}
-
 		// By setting this annotation, the CDI will set ownership of the PVC to the DV
 		pvc.Annotations[populatedForPVCAnnotation] = dvOwner
 	} else if !isVolumeOwnershipPolicyNone(vmRestore) { // PVC is owned by the VM
-		target.Own(pvc)
+		if target.Exists() {
+			target.Own(pvc)
+		} else if len(volumeBackup.PersistentVolumeClaim.OwnerReferences) > 0 {
+			pvc.Annotations[restoreOwnedByVMLabel] = "true"
```

Comment:
Why can't the ownerreferences be set here? Restoring to a new VM? Seems like there may have been a bug with previous implementation then? 

If this annotation is necessary, I think it would be better to use the name of the restore instead of "true"

Reply:
Because the VM doesn't exist yet at this point (aka target doesn't contain the VM). And thus we don't have the UID of the VM.